packages/design-system/src/components/TextField/LabelMask/LabelMask.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useRef } from 'react';
+
+export type LabelMaskMask = 'phone' | 'ssn' | 'date' | 'day_month';
+
+export interface LabelMaskProps {
+  /**
+   * Must contain a `TextField` component
+   */
+  children: React.ReactNode;
+  labelMask?: LabelMaskMask | ((rawInput: string) => string);
+  // placeholder?: string;
+  // value?: string;
+}
+
+const LabelMask = ({ children, labelMask }: LabelMaskProps) => {
+  const labelMaskPattern = {
+    phone: '[0-9-]*',
+    ssn: '[0-9-*]*',
+    date: '[0-9-]*',
+    day_month: '[0-9.,-]*',
+  };
+
+  /**
+   * Get the child text field. Called as a method so that
+   * updates to the field cause the mask to re-render
+   * @returns {React.ReactElement} Child TextField
+   */
+  const field = (): React.ReactElement => React.Children.only(children as React.ReactElement);
+  const initialValue = field().props.value || field().props.defaultValue || '### ### ###';
+
+  const [value, setValue] = useState(initialValue);
+  const [focus, setFocus] = useState(null);
+
+  /**
+   * @param {Object} evt
+   * @param {React.Element} field - Child TextField
+   */
+  const handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+    // Eval use of state here
+    setValue(evt.target.value);
+    // if (typeof field().props.onChange === 'function') {
+    //   value = field().props.onChange(evt);
+    // }
+  };
+
+  const textRef = useRef();
+  const modifiedTextField = React.cloneElement(field(), {
+    defaultValue: undefined,
+    onChange: (e) => handleChange(e),
+    onBlur: () => setFocus(false),
+    onFocus: () => setFocus(true),
+    type: 'text',
+    inputMode: 'numeric',
+    ref: textRef,
+    // pattern: labelMaskPattern[labelMask],
+  });
+
+  return (
+    <div className={`ds-c-field-mask ds-c-field-mask--${labelMask}`}>
+      {/* mask in render */}
+      {/* {formatValue()} */}
+      {focus ? <span>{value}</span> : <span>(xxx) xxx-xxxx</span>}
```

Comment:
What I was meaning here was that we should let CSS determine if we're focused or not

```suggestion
<span className="ds-c-field-labelmask__mask--active">{labelMask(value)}</span>
<span className="ds-c-field-labelmask__mask--inactive">{labelMask('')}</span>
```

```css
.ds-c-field-labelmask__mask--active {
  display: none;
  &:focus {
    display: block;
  }
}

.ds-c-field-labelmask__mask--inactive {
  display: block;
  &:focus {
    display: none;
  }
}
```
This way we don't have to use any state to determine if this element has focus within JavaScript

Reply:
This isn't working because `<span>` isn't a focusable element. 

The HTML structure is kinda like this:
```html
<div>
    <div class="label-mask>
        <span class="active">Foo</span>
        <span class="active">Bar</span>
    </div>
    <input class="text-input" />
</div>
```

I attempted using sibling selectors to target if the input has focus, but [this selector relies on the structure of the markup](https://developer.mozilla.org/en-US/docs/Web/CSS/Adjacent_sibling_combinator). Since the label comes before the input (both visually and structurally), I'm unable to target something like 

```css
.text-input:focus + .label-mask .active { display: inline-block; }
.text-input + .label-mask .active { display: none; }
```

afaik the only way to target the sibling of a focusable element, regardless of DOM order, is via the `:has` attribute, which is only available in Safari [sad trombone].

Reply:
Alternatively... I might be able to change the visual appearance of the DOM order using `flex` and `order`, which would allow the sibling selector to work - but it _might_ be a [WCAG violation](https://www.w3.org/TR/WCAG20-TECHS/C27.html). I say might because we are intending to hide the label mask from AT anyways, so I'm not entirely sure it matters if it's visually before or after... 🤔 

I'm going to attempt this and discuss with a11y folks.

Reply:
Ah, that's too bad 😞